src/elements/ProgressBar/ProgressBar.js

```diff
@@ -44,20 +46,17 @@ export default class ProgressBar extends React.Component {
 	}
 
 	componentWillUnmount() {
-		if(this.state.interval) {
-			clearInterval(this.state.interval);
-		}
+		clearInterval(this.interval);
 	}
 
 	timer() {
-		this.setState({ percentage: this.state.percentage + 1 });
-
-		if (this.state.percentage === 100) {
-			clearInterval(this.state.interval);
+		this.setState({
+			percentage: this.state.percentage + 1
+		});
 
-			if (this.props.onComplete) {
-				setTimeout(this.props.onComplete, this.props.completeDelay);
-			}
+		if (this.state.percentage >= 100) {
```

Comment:
Don't forget to add this logic inside of a callback function to the above `setState`:

```
setState({....}, () => {
  if(this.state......)
});